DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs

```diff
@@ -1738,6 +1760,77 @@ private void HandleMapSharingBlockedMessage(string sender)
                 "they will be unable to participate in this match.".L10N("UI:Main:PlayerMissingMaDisabledSharing"), sender));
         }
 
+        /// <summary>
+        /// Download a map from CNCNet using a map hash ID.
+        ///
+        /// Users and testers can get map hash IDs from this URL template:
+        ///
+        /// - http://mapdb.cncnet.org/search.php?game=GAME_ID&search=MAP_NAME_SEARCH_STRING
+        ///
+        /// </summary>
+        /// <param name="parameters">
+        /// This is a string beginning with the sha1 hash map ID, and (optionally) the name to use as a local filename for the map file.
+        /// Every character after the first space will be treated as part of the map name.
+        ///
+        /// "?" characters are removed from the sha1 due to weird copy and paste behavior from the map search endpoint.
+        /// </param>
+        private void DownloadMapByIdCommand(string parameters)
+        {
+            string sha1;
+            string mapName;
+            string message;
+
+            // Check if the parameter's contain spaces.
+            // The presence of spaces indicates a user-specified map name.
+            parameters = parameters.Trim();
+            int firstSpaceIndex = parameters.IndexOf(' ');
```

Comment:
Wouldn't this fail if a user supplied one word as an argument?

Reply:
It returns `-1` if there isn't a space. 

In my tests I was usually doing `/downloadmap HASH` and `/downloadmap HASH onewordname` and it seemed to work.